modulus/models/layers/fused_silu.py

```diff
@@ -13,16 +13,27 @@
 # limitations under the License.
 
 import functools
+import logging
+from typing import Tuple
 
 import torch
 from torch.autograd import Function
 
+logger = logging.getLogger(__name__)
+
 try:
-    from nvfuser._C import DataType, Fusion, FusionDefinition
-except ImportError:
-    # accomodating for earlier versions of PyTorch (< 2.0)
-    # which don't need nvfuser as explicit dependency
-    from torch._C._nvfuser import DataType, Fusion, FusionDefinition
+    import nvfuser
+    from nvfuser import DataType, FusionDefinition
+except ImportError as e:
```

Comment:
Does `logger.error` re-raise the exception `e`?

Reply:
I am not sure if I quite understand. This currently prints a message like below:

```
>>> from modulus.models.layers.fused_silu import silu_backward_for
An error occured. Details: cannot import name 'DataType' from 'nvfuser' (/usr/local/lib/python3.10/dist-packages/nvfuser/__init__.py) 
Either nvfuser is not installed or the version is incompatible. Please retry after installing correct version of nvfuser. The new version of nvfuser should be available in PyTorch container version >= 23.10. https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/index.html
```

Reply:
If you don't re-raise the exception, then you'll only get this error message and not any of the trace. It's better to do something like
```python
try:
    from nvfuser import DataType, FusionDefinition
except ImportError as e:
    logging.error(...)
    raise  # Only a raise without an exception will re-raise the last exception
```

Reply:
Gotcha, added